Fetch summary for daily Darksky forecasts

leppa opened the pull request:
## Description:

Dark Sky service provides a summary for each forecast day. This (very simple) change will introduce a new entity `sensor.dark_sky_summary_<day>` for each corresponding day listed in `forecast` parameter.

## Example entry for `configuration.yaml`:
```yaml
- platform: darksky
  forecast:
    - 1
  monitored_conditions:
    - summary
```

## Checklist:
  - [X] The code change is tested and works locally.

leppa committed:
darksky: Fetch summary for daily forecasts

kellerza reviewed (commented):
This seems like the intended design reading the docs

What happens if you change forecast to something other than 1 (say 5)?

leppa commented:
What do you mean by that?

Dark Sky returns weekly summary in daily forecast, as well as daily summary for each day:

```json
{
    "daily": {
        "summary": "Week summary",
        "data": [{
            "summary": "Day summary"
        }],
    }
}
```

The former was exposed by Home Assistant, the latter wasn't. This change exposes it. If you specify:

```yaml
- platform: darksky
  forecast:
    - 1
    - 4
  monitored_conditions:
    - summary
```

you will get the following entities: `sensor.dark_sky_summary` (week summary), `sensor.dark_sky_summary_1` (summary for day 1), and `sensor.dark_sky_summary_4` (summary for day 4).

kellerza commented:
Thats the idea, but do you see it wokring like that?

leppa commented:
I'm pretty sure it does. I'll check tomorrow and let you know.

leppa commented:
I've checked today. With:

```yaml
- platform: darksky
  forecast:
    - 4
  monitored_conditions:
    - summary
```

you'll get `sensor.dark_sky_summary_4` with the summary of day 4.

kellerza commented:
Still seems like you are (accidentally?) fixing something that was introduced earlier (#5264).

Would be good to hear from someone using this functionality... I see @fabaff is code owner, maybe he has a better idea of what is happening here

leppa commented:
I feel like we're having a misunderstanding. There was and still is (see https://darksky.net/dev/docs#/dev/docs#response-format):
  - `minutely_summary`, a ["human-readable summary of this **data block**"](https://darksky.net/dev/docs#data-block) for the next hour. Extracted from `minutely.summary` property in JSON.
  - `hourly_summary`, a "human-readable summary of this **data block**" for the next two days. Extracted from `hourly.summary` property in JSON.
  - `daily_summary`, a "human-readable summary of this **data block**" for the next week. Extracted from `daily.summary` property in JSON.
  - `summary`, which is a ["human-readable text summary of this **data point**"](https://darksky.net/dev/docs#data-point) for day 0 (today). Extracted from `daily.data[0].summary` property in JSON.

However, there was no way to get a "human-readable text summary of this **data point**" for days 1 to 7, while other properties of these days (like temperature, humidity, wind, etc.) were available. My change makes it possible to also get a summary.

kellerza commented:
Thx, clear now and I retested with some results. Maybe you can try to clarify in the official docs?
